Source/Diagnostics/ReducedDiags/ParticleHistogram2D.cpp

```diff
@@ -0,0 +1,302 @@
+
+                long const np = pti.numParticles();
+
+                //Flag particles that need to be copied if they cross the z_slice
+                amrex::ParallelFor(np,
+                                   [=] AMREX_GPU_DEVICE(int i)
+                                   {
+                                       amrex::ParticleReal x, y, z;
+                                       GetPosition(i, x, y, z);
+                                       auto const w  = (amrex::Real)d_w[i];
+                                       auto const ux = d_ux[i] / PhysConst::c;
+                                       auto const uy = d_uy[i] / PhysConst::c;
+                                       auto const uz = d_uz[i] / PhysConst::c;
+
+                                       // don't count a particle if it is filtered out
+                                       if (do_parser_filter)
+                                           if (!fun_filterparser(t, x, y, z, ux, uy, uz, w))
+                                               return;
+
+                                       // continue function if particle is not filtered out
+                                       auto const f_abs = fun_partparser_abs(t, x, y, z, ux, uy, uz, w);
+                                       auto const f_ord = fun_partparser_ord(t, x, y, z, ux, uy, uz, w);
+                                       auto const weight = fun_valueparser(t, x, y, z, ux, uy, uz, w);
+
+                                       // determine particle bin
+                                       int const bin_abs = int(Math::floor((f_abs-bin_min_abs)/bin_size_abs));
+                                       if ( bin_abs<0 || bin_abs>=num_bins_abs ) return; // discard if out-of-range
+
+                                       int const bin_ord = int(Math::floor((f_ord-bin_min_ord)/bin_size_ord));
+                                       if ( bin_ord<0 || bin_ord>=num_bins_ord ) return; // discard if out-of-range
+
+                                       amrex::Real &data = d_table(bin_abs, bin_ord);
+                                       amrex::HostDevice::Atomic::Add(&data, weight);
+                                   });
+            }
+        }
+    }
+
+    // Copy data from GPU memory
+    m_h_data_2D.copy(d_data_2D);
+
+    // reduced sum over mpi ranks
+    int size = static_cast<int> (d_data_2D.size());
+    ParallelDescriptor::ReduceRealSum
+            (h_table_data.p, size, ParallelDescriptor::IOProcessorNumber());
+
+    // Return for all that are not IO processor
+    if ( !ParallelDescriptor::IOProcessor() ) { return; }
+}
+// end void ParticleHistogram2D::ComputeDiags
+
+void ParticleHistogram2D::WriteToFile (int step) const
+{
+#ifdef WARPX_USE_OPENPMD
+    // only IO processor writes
+    if ( !ParallelDescriptor::IOProcessor() ) { return; }
+
+    // Create the OpenPMD series
+    auto series = io::Series(
+            m_path + "hist2D/openpmd_%06T." + m_openpmd_backend,
```

Comment:
Let's check what is in `m_path` so far:
We need to potentially add an additional folder between `m_path` and `hist2D`, with the name of the reduced diagnostics `<reduced_diags_name>`, so we can have multiple `hist2D` active in the same simulation, e.g., for a longitudinal and a transversal phase space.